Assets/00_Content/Scripts/Scenes/SceneLoadManager.cs

```diff
@@ -78,20 +79,47 @@ public class SceneLoadManager : SingletonBehaviour<SceneLoadManager> {
 					DontDestroyOnLoad(player.gameObject);
 			}
 
+			SceneInfo currentScene = CurrentScene;
+			MusicCue nextMusic = sceneInfo.music;
+			bool switchMusic = nextMusic != currentScene?.music;
+
 			void BeforeEnterScene() {
 				beforeEnterScene?.Invoke();
 
+				if (switchMusic)
+					AudioManager.PlayMusicSafe(nextMusic, FadeKind.OutIn, transition.TransitionDuration / 2f);
+
 				// Reset time scale when loading a new scene
 				Time.timeScale = 1f;
 			}
 
 			if (doTransition) {
+				if (switchMusic)
+					AudioManager.StopMusicSafe(transition.TransitionDuration / 2f);
+
 				transition.Play(sceneInfo, BeforeEnterScene);
 			}
 			else {
+				// FIXME: How do we fade here?
```

Comment:
I'm guessing that this is just a note for you until later? Or is this something that actually should be done before this pull request is approved?

Reply:
Nah, it's for later. Currently this case is not even used.